### PR TITLE
Discourage use of Timecop.(freeze/travel) w/o a block

### PR DESCRIPTION
**Pull request by kchasel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -412,6 +412,40 @@ which should be validated. Using `be_valid` does not guarantee that the problem
       end
     end
     ```
+        
+* Avoid use of Timecop in a spec as often as possible. If necessary, always use
```
Comment (oggy) [inline README.md]: I'm -1 on the first sentence unless you mean that it should be stubbed across the entire test suite instead. And if you do this, then the second part is largely unnecessary.

**Comment by kchasel** [comment]
@oggy I do think stubbing across the entire suite is a good idea. However, if you use Timecop within a spec, even when time is globally stubbed in the suite, and the Timecop code inside the spec doesn't properly return, this will adversely affect the rest of the suite.

**Comment by oggy** [comment]
@kchasel Okay, so maybe something just to clarify that point then?

Also, I'm not sure the example reflects what you're saying. You're talking about not doing this, right?

```
it "does something futuristic" do
  Timecop.travel(1.year.from.now)
  ...
end
```

Using `before` and `after` blocks works fine as far as I can tell, and is useful if you want to affect all the specs in an example group.

**Comment by kchasel** [comment]
@oggy technically yes, it works fine. However, we've repeatedly run into issues on couples with that form failing to return (for whatever reason) and affecting the other specs (this morning I solved some rspec failures we were having with a new spec by modifying a spec that runs before it in the suite to no longer use `Timecop.return` in an `after { ... }`

**Comment by namxam** [comment]
I agree with @kchasel. Using Timecop with `before` and `after` blocks works, but almost every project I worked on, had order dependent failures due to missing `Timecop.return` blocks. I would encourage everybody to use the block syntax with an `around` block.

**Comment by kchasel** [comment]
@oggy @igkins :thought_balloon:s:question:

**Pull request closed without merging**